openmeter/app/entity/app.go

```diff
@@ -21,6 +23,52 @@ type App interface {
 
 	// ValidateCapabilities validates if the app can run for the given capabilities
 	ValidateCapabilities(capabilities ...appentitybase.CapabilityType) error
+
+	// Customer data
+	GetCustomerData(ctx context.Context, input GetAppInstanceCustomerDataInput) (CustomerData, error)
+	UpsertCustomerData(ctx context.Context, input UpsertAppInstanceCustomerDataInput) error
+	DeleteCustomerData(ctx context.Context, input DeleteAppInstanceCustomerDataInput) error
+}
+
+type GetAppInstanceCustomerDataInput struct {
```

Comment:
nit:
```suggestion
type GetAppInstanceCustomerDataInput = customerentity.CustomerID
```

This would save you some unnecessary typecasts.